src/Infrastructure/Persistence/ParticipantDynamoDbRepository.cs

```diff
@@ -68,7 +73,7 @@ public async Task<ParticipantDetails> QueryIndexForParticipantDetailsAsync(strin
                 _logger.LogInformation("item: {Item}", JsonConvert.SerializeObject(item, Formatting.Indented));
 
                 var participant = _context.FromDocument<ParticipantDetails>(Document.FromAttributeMap(item));
-                return await GetParticipantDetailsAsync(participant.Pk.Replace("PARTICIPANT#", ""));
+                return await GetParticipantDetailsAsync(participant.Pk.Replace("PARTICIPANT#", ""), cancellationToken);
             }
             catch (Exception e)
```

Comment:
No need for this `try-catch` if you are going to output and rethrow.
Your debugger should break when the exception is first thrown.